packages/block-library/src/embed/embed-placeholder.native.js

```diff
@@ -40,12 +44,58 @@ const EmbedPlaceholder = ( {
 	);
 	const embedIconErrorStyle = styles[ 'embed__icon--error' ];
 
+	const cannotEmbedMenuPickerRef = useRef();
+
+	const errorPickerOptions = {
+		retry: {
+			id: 'retryOption',
+			label: __( 'Retry' ),
+			value: 'retryOption',
+			onSelect: () => {
+				tryAgain();
+			},
+		},
+		convertToLink: {
+			id: 'convertToLinkOption',
+			label: __( 'Convert to link' ),
+			value: 'convertToLinkOption',
+			onSelect: () => {
+				fallback();
+			},
+		},
+	};
+
+	const options = compact( [
+		cannotEmbed && errorPickerOptions.retry,
+		cannotEmbed && errorPickerOptions.convertToLink,
+	] );
+
+	function onPickerSelect( value ) {
+		const selectedItem = options.find( ( item ) => item.value === value );
+		selectedItem.onSelect();
+	}
+
+	function onPickerPresent() {
+		if ( cannotEmbedMenuPickerRef.current ) {
+			cannotEmbedMenuPickerRef.current.presentPicker();
+		}
```

Comment:
We could simplify this block with the following line:
```suggestion
		cannotEmbedMenuPickerRef.current?.presentPicker();
```

I also see that this function (`onPickerPresent`) is only called from `resolveOnPressEvent` so we might consider removing it and move the code there, wdyt?

Reply:
Great observation! Done here deb2eb7